qa/rpc-tests/feature_blocksign.py

```diff
@@ -0,0 +1,114 @@
+
+class BlockSignTest(BitcoinTestFramework):
+
+    # Dynamically generate 3 keys to be used for block signing.
+    def init_keys(self, num_keys):
+        self.keys = []
+        self.wifs = []
+        for i in range(num_keys):
+            k = CECKey()
+            pk_bytes = hashlib.sha256(str(random.getrandbits(256)).encode('utf-8')).digest()
+            k.set_secretbytes(pk_bytes)
+            w = wif(pk_bytes)
+            print("generated key {}: \n  priv: {}\n  pub: {}\n  wif: {}".format(i+1,
+                k.get_privkey().hex(),
+                k.get_pubkey().hex(),
+                w))
+            self.keys.append(k)
+            self.wifs.append(wif(pk_bytes))
+
+    # The signblockscript is a Bitcoin Script 3-of-3 multisig script.
+    def make_signblockscript(self):
+        # making 3-of-3 multisig (full quorum)
+        n = 3
+        script = "{}".format(50+n) # full quorum
+        for i in range(n):
+            k = self.keys[i]
+            script += "41"
+            script += k.get_pubkey().hex()
+        script += "{}".format(50+n) # num keys
+        script += "ae" # OP_CHECKMULTISIG
+        return script
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 3
+        self.init_keys(3)
+        self.extra_args = [[
+            "-chain=blocksign",
+            # We can't validate pegins since we don't run the parent chain.
+            "-validatepegin=0",
+            "-signblockscript={}".format(self.make_signblockscript())
+        ]] * 3
+
+    def setup_network(self, split=False):
+        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir, self.extra_args)
+        # Have every node import its block signing private key.
+        for i in range(3):
+            self.nodes[i].importprivkey(self.wifs[i])
+        connect_nodes_bi(self.nodes, 0, 1)
+        connect_nodes_bi(self.nodes, 1, 2)
+        connect_nodes_bi(self.nodes, 0, 2)
+        self.is_network_split = False
+        self.sync_all()
+
+    def mine_block(self):
+        # mine block in round robin sense: depending on the block number, a node
+        # is selected to create the block, others sign it and the selected node
+        # broadcasts it
+        mineridx = self.nodes[0].getblockcount() % 3 # assuming in sync
+        miner = self.nodes[mineridx]
+
+        # miner makes a block
+        block = miner.getnewblockhex()
+        # all nodes sign
+        sigs = []
+        for n in self.nodes:
+            sigs.append(n.signblock(block))
+
+        # miner submits
+        result = miner.combineblocksigs(block, sigs)
+        assert_equal(result["complete"], True)
+        signedblock = result["hex"]
+        miner.submitblock(signedblock)
+
+    def mine_blocks(self, num_blocks):
+        for i in range(num_blocks):
+            self.mine_block()
+            self.sync_all()
+
+    def run_test(self):
+        # Check that there's 100 UTXOs on each of the nodes
```

Comment:
I don't think these checks are useful in this test. Check the block height instead?